src/pages/UserAccount/User.js

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import './SignUp.scss';
+
+function User({ userData, isSelectLogin }) {
+  const { title, text, url, button } = userData;
+  const [inputValue, setInputValue] = useState({ id: '', pw: '' });
+  const navigate = useNavigate();
+  const [validId, setValidId] = useState(false);
+  const [validPw, setValidPw] = useState(false);
+  const [termsAgree, setTermsAgree] = useState(false);
+
+  const handleInput = e => {
+    const { name, value } = e.target;
+    setInputValue({ ...inputValue, [name]: value });
+  };
+
+  // const data = { id: 'abc@', pw: '123' };
+  useEffect(() => {
+    const regExp = /^(?=.*\d)(?=.*[a-zA-Z])[0-9a-zA-Z]{8,16}$/;
+    if (!regExp.test(inputValue.id)) {
+      setValidId(false);
+    } else {
+      setValidId(true);
+    }
+  }, [inputValue.id]);
+
+  useEffect(() => {
+    const regExp =
+      /^(?=.*[a-zA-z])(?=.*[0-9])(?=.*[$`~!@$!%*#^?&\\(\\)\-_=+]).{8,16}$/;
+    if (!regExp.test(inputValue.pw)) {
+      setValidPw(false);
+    } else {
+      setValidPw(true);
+    }
+  }, [inputValue.pw]);
+
+  const validator = validId === true && validPw === true && termsAgree === true;
```

Comment:
```suggestion
  const idRegExp = /^(?=.*\d)(?=.*[a-zA-Z])[0-9a-zA-Z]{8,16}$/;
  const pwRegExp =
    /^(?=.*[a-zA-z])(?=.*[0-9])(?=.*[$`~!@$!%*#^?&\\(\\)\-_=+]).{8,16}$/;

  const validId = idRegExp.test(inputValue.id);
  const validPw = pwRegExp.test(inputValue.pw);

  const validator = validId && validPw && termsAgree;
```

위에서 말씀 드린 것 처럼 이처럼 state가 변경 될 때마다 리렌더링을 통해
변수값이 계산되기 때문에 위와같이 관리가 가능해집니다!